### ISSUE TITLE
Questions: Performance: Ram memory scaling in Java processes, and cuda acceleration 

### ISSUE THREAD

**Issue by dorvan** [mzmine/mzmine2#334]
Hello, 

I'm working on custom tuned system (hardware + custom linux) for mzmine and R based software speedup.

- The system it's fully recompiled and updated;
- I have tuned all java processes parameters in the mzmine launche script, benchmarking results;
- the system have no bootlenecks (compiled native, 64gb ram, 8threads, nvme 10gbit/s nvme disk);
- Nvidia Tesla K10 system available with nvidia and mesa mapping;
- Latest Oracle JVM (SDK & JRE) *.144 updated on 28/07/2017;
- Actually Intel cpu architecture, Linux system.

MZmine works on:
- nvme disk 10,4gbit/s, 1,3gbyte/s (real);
- 64gbyte ram total (<500mbyte used by system) 60gbyte dedicated to mzmine 25gbit/s, 3.2Gbyte/s (real).

MZmine java process Params (best performance and scaling):
xms: 60Gbyte
xmm: 60Gbyte (HIPE)
temp and work dir on nvme disk.

In this scenario I see the same behavior of all other installations of mzmine, but "accelerated" (obviously).

Where, calculation starts and allocate resources, Ram Memory and process Threads, scale up, and after completing first files (in seconds) other files takes more and more time to calculate in a "relaxed" situation taking minutes to complete last files. This behaviour for all kind of calculations.
and can be simply monitored with an "htop" (for memory and threads) and all "top" software like, "iotop" (to see I/O) 

cpu allocations:

- **on file loading (n.8):** 1/8 threads at 100%, other at 0%, java already forked in threads for files
, **load average 1.x**.
- **on files calculation starts:** 8/8 threads at 100% for thread management, not for calcuation complexity, cores are allocated but not stressed, only 1 o 2 cores are stressed at same time (rotating), **load average 5.x - 7.x**
- **on calculation, ( after some seconds, 4 files ends calculation, in small amount of time diff):**  4/8 threads at 100%, 4 at 0% (rotating) for thread management, not for calcuation complexity, cores are allocated but not stressed, 0,5 cores are stressed (rotating), **load average 2.x**
- 4 files ending in seconds, 3 files files ending in minutes, 1 file is x5 slowest to end compared the other files, but calculation pressure and allocation going down quickly, and machine calculate in a totally relaxed situation, more on each close file. On a machine like this, can be filled up with a load average of 12-15-17, considering I/O.

My questions are:

1. What is the best way to tune up mzmine to scale up, more than I have already done? 
2. Anyone match this MZmine behaviour (on any kind of machine and os) like me?
3. MZmine have particular settings to scale up more?
4. How I can use R software asset to scale up MZmine today?
5. MZmime today, can bind on nvidia cuda or opencl or opengl or mesa to scale up on GPU?
6. seems on Big Pressure(some files to calculate), queue and resources are managed well, can be a queue management problem?

Thanks.
and Thanks all for your works on MZmine.

**Comment by tomas-pluskal** [comment]
Hi,
1. Improve the code of the MZmine algorithms - many of the algorithms are rather simple and could use optimization
2. No idea
3. Not really. You can adjust the number of tasks running in parallel in the MZmine preferences.
4. No idea
5. Not that I know of
6. I don't understand the question

**Comment by dorvan** [comment]
@tomas-pluskal Thanks for answer.
3: where exactly in configuration file?
6: the MZmine behaviour seems a queue management problem, about the task queue.

**Comment by tomas-pluskal** [comment]
3. Just open the Preferences dialog when you run MZmine
6. I still don't understand what your question is
